### ISSUE TITLE
Knative Serving service shows IngressNotConfigured and unable to access the application via Kong Ingress

### ISSUE THREAD

**Issue by MaharaJothi** [Kong/kubernetes-ingress-controller#2964]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I've set up Knative Serving in EKS cluster with Kong configured for Ingress gateway.
Below are the cluster details:
Kubernetes Version: 1.23
Knative Version: 1.7.1
Kong Version: 2.6.9

Installed Knative using:
```
kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.7.1/serving-crds.yaml
kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.7.1/serving-core.yaml
```

Updated Domain name:
```
kubectl patch configmap/config-domain \
  --namespace knative-serving \
  --type merge \
  --patch '{"data":{"test.com":""}}'
```

Updated ingress class:
```
kubectl patch configmap/config-network \
  --namespace knative-serving \
    --type merge \
      --patch '{"data":{"ingress.class":"prod-kong-ingress"}}'
```

Deployed the application by following [this](https://knative.dev/docs/serving/services/creating-services/#procedure).

After deploying the application yaml, `kubectl get ksvc` displays the URL in which application can be accessible.  But `READY` shows `Unknows` and `REASON` as `IngressNotConfigured`

```
NAME    URL                             LATESTCREATED   LATESTREADY   READY     REASON
hello   http://hello.default.test.com   hello-00001     hello-00001   Unknown   IngressNotConfigured
```
While accessing the application URL, am seeing the below:
`{"message":"no Route matched with those values"}`

It looks like the K8s service is not being assigned to the ingress.  
But I could see the below annotation available in K8s service created by Knative:
`networking.knative.dev/ingress.class: prod-kong-ingress`

Can someone help me identify the cause and fix this?
Am I updating the ingress class properly?

### Expected Behavior

`READY` should be `True` and the application URL should be accessible.

### Kong Ingress Controller version

```shell
v2.6.0
```

### Kubernetes version

```shell
1.23
```

**Comment by MaharaJothi** [comment]
I've reached out to [Knative community](https://github.com/knative/serving/issues/13325) but I've been redirected here

**Comment by pmalek** [comment]
Hi @MaharaJothi 👋 

> Kong Ingress Controller version v2.6.9

Are you sure you didn't mean `v2.6.0`? Or you meant Kong version `v2.6.9` as in the `Current Behavior` section?

**Comment by MaharaJothi** [comment]
Hi @pmalek , Yes It's 2.6.0.  It was a typo from my end which I corrected. Thanks for pointing it out.

**Comment by pmalek** [comment]
OK, got it.

#2543 might be the same issue but we'd need to take a closer, in depth look at it to verify.

Please note, that as mentioned in https://github.com/Kong/kubernetes-ingress-controller/issues/2543#issuecomment-1195910238, Knative support in KIC is still behind a feature flag (as per https://github.com/Kong/kubernetes-ingress-controller/blob/main/FEATURE_GATES.md#feature-gates-for-alpha-or-beta-features) and still marked as Alpha.

While we're trying to account for all the reports from our users we might be lacking resources to address all of them, especially those for alpha/beta features behind feature gates.

This will, as all the other issues be accounted for in our project's board but as mentioned above it might end up having lower priority than other issues.

Let us know if the above makes sense and also please be aware that we always welcome contributions from outside contributors.

**Comment by MaharaJothi** [comment]
Thanks for the update @pmalek 

I was able to make it work. 
Earlier I installed Kong and then Knative.  Whenever I install in this order, I got the issue I have reported.

However, if I install Knative first and then Kong, it all seems to work fine without any issue.
I notice the same order has been mentioned in kong documentation as well: https://docs.konghq.com/kubernetes-ingress-controller/2.6.x/guides/using-kong-with-knative/
